eslint.config.mjs

```diff
@@ -0,0 +1,391 @@
+          message:
+            'callWhen wraps the function in a promise; instead immediately return a vow',
+        },
+        {
+          selector: "Identifier[name='heapVowE']",
+          message:
+            'heapVowE shortens vows to promises; instead use `E` from `@endo/far` with `watch` from durable vowTools',
+        },
+        {
+          selector: "Identifier[name='heapVowTools']",
+          message:
+            'heapVowTools are not durable; instead use `prepareVowTools` with a durable zone',
+        },
+      ],
+    },
+  },
+  {
+    // Allow "loan" contracts to mention the word "loan".
+    files: ['packages/zoe/src/contracts/loan/*.js'],
+
+    rules: {
+      'no-restricted-syntax': ['error', ...deprecatedTerminology.loanContract],
+    },
+  },
+  {
+    files: ['**/*.ts'],
+
+    rules: {
+      'jsdoc/require-param-type': 'off',
+      'no-undef': 'off',
+    },
+  },
+  {
+    files: ['**/*.d.ts'],
+
+    rules: {
+      'no-redeclare': 'off',
+    },
+  },
+  ...compat
+    .extends('plugin:@typescript-eslint/disable-type-checked')
+    .map(config => ({
+      ...config,
+      files: [
+        '**/exported.*',
+        '**/types-index.*',
+        '**/types-ambient.*',
+        '**/types.*',
+      ],
+    })),
+  {
+    files: ['**/*.html'],
+
+    languageOptions: {
+      ecmaVersion: 5,
+      sourceType: 'script',
+
+      parserOptions: {
+        project: false,
+      },
+    },
+  },
+  ...compat
+    .extends('plugin:@typescript-eslint/disable-type-checked')
+    .map(config => ({
+      ...config,
+      files: ['a3p-integration/**'],
+    })),
+  {
+    files: ['a3p-integration/**'],
+
+    languageOptions: {
+      ecmaVersion: 5,
+      sourceType: 'script',
+
+      parserOptions: {
+        useProjectService: false,
+        project: false,
+      },
+    },
+
+    rules: {
+      '@jessie.js/safe-await-separator': 'off',
+    },
+  },
+  {
+    files: ['**/types*.js'],
+
+    rules: {
+      '@typescript-eslint/no-floating-promises': 'off',
+    },
+  },
+  {
+    // Rules specific to swingset-xsnap-supervisor
+    files: ['packages/swingset-xsnap-supervisor/**/*.js'],
+    rules: {
+      'import/no-extraneous-dependencies': 'off',
+    },
+  },
+  // Separate config for multichain-testing that doesn't inherit root rules
+  {
+    files: ['multichain-testing/**/*.{js,ts,mjs,cjs}'],
+    ignores: [], // Clear any ignores from previous configs
+    languageOptions: {
+      parser: typescriptEslint.parser,
+      parserOptions: {
+        ecmaVersion: 'latest',
+        sourceType: 'module',
+        project: './multichain-testing/tsconfig.json',
+      },
+    },
+    // Only enable the no-unused-vars rule
+    rules: {
+      '@typescript-eslint/no-unused-vars': [
+        'error',
+        {
+          argsIgnorePattern: '^_',
+          varsIgnorePattern: '^_',
+        },
+      ],
+      // Disable all other rules
+      '@typescript-eslint/no-floating-promises': 'off',
```

Comment:
I want to see these suppressions omitted but if we do end up keeping them please use "warn" instead of disabling them completely. They should still give notice to developers even if they don't fail CI.